File: problems/0009_palindrome-number/reversed-int/Solution.scala
object Solution {
    def isPalindrome(x: Int): Boolean = {
        if (x < 0)
            return false
        else if (x < 10)
            return true
        var (reversed, k) = (0, x)
        while (k != 0) {
            reversed = reversed * 10 + (k % 10)
            k /= 10
        }
        x == reversed
    }
}

File: problems/0009_palindrome-number/reversed-int/Solution.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0)
            return false;
        else if (x < 10)
            return true;
        long reversed = 0;
        for (auto k = x; k != 0; k /= 10)
            reversed = reversed * 10 + (k % 10);
        return x == reversed;
    }
};
